File: dotnet-core/Tea/Actions/Action.cs
using CustomExtensions;
using OpenQA.Selenium;

namespace Tea
{
    public abstract class Action
    {
        protected By _by;
        public Action(By by) => _by = by;
        public abstract void Perform(WebBrowser browser, string? value);
    }

    public class Click : Action
    {
        
        public Click(By by) : base(by) {}
        public override void Perform(WebBrowser browser, string? value)
        {
            Element element = browser.FindElement(_by);
            element.Click();
        }
    }

    public class TypeText : Action
    {
        public TypeText(By by) : base(by) {}
        public override void Perform(WebBrowser browser, string value)
        {
            if ( value == "random")
            {
                browser.GetActiveElement().TypeText(value.GetRandomString(10).Replace("\"", ""));

            } else {
                browser.GetActiveElement().TypeText(value.Replace("\"", ""));
                
            }
        }
    }
}

File: java/tea-framework/src/main/java/com/tea/framework/Actions/Action.java
package com.tea.framework.Actions;

import com.tea.framework.SeleniumWrapper.Drivers.WebBrowser;
import org.openqa.selenium.By;

import java.nio.charset.Charset;
import java.util.Random;

public abstract class Action {
    protected By _by;

    public Action(By by) {
        _by = by;
    }

    public abstract void Perform(WebBrowser browser, String value);

    protected String GetRandomString(int i) {
        byte[] a = new byte[i];
        new Random().nextBytes(a);
        return new String(a, Charset.forName("UTF-8"));
    }
}
